src/aspnetcore/src/SignalR/server/Core/src/Internal/HubMethodDescriptor.cs

```diff
@@ -97,7 +97,7 @@ public HubMethodDescriptor(ObjectMethodExecutor methodExecutor, IServiceProvider
                         {
                             if (keyedServiceProvider.IsKeyedService(GetServiceType(p.ParameterType), keyedServicesAttribute.Key))
                             {
-                                KeyedServiceKeys ??= new List<(int, object)>();
+                                KeyedServiceKeys ??= new List<(int, object?)>();
```

Comment:
@halter73 are you going to need to do anything different here to honor https://github.com/dotnet/dotnet/blob/22fe462b4755376d90dc942f85e103abe060d16e/src/runtime/src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/FromKeyedServicesAttribute.cs#L44-L47

Reply:
I don't think so since we're the first level of services as far as DI is concerned. So there would be no parent to inherit the key from, and my reading is that `null` will be the key value in that case which will grab the non-keyed service from DI as expected?

Same with Minimal APIs and MVC, I think we're considered the first level of services.

Not sure about Blazor though, maybe they have multiple layers of DI and might want to react here?

Reply:
It looks like these changes were enough to make the build work, so let's move forward with them.  Would you like me to file an issue to follow up with testing parts of ASP.NET with the different types of `ServiceKeyLookupMode`?

Reply:
I went ahead and opened https://github.com/dotnet/aspnetcore/issues/62054 to track that.